src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -584,6 +588,21 @@ void MapCommonStreamConfigs(auto &memory, StreamQuery &stream_query) {
 }
 }  // namespace
 
+antlrcpp::Any CypherMainVisitor::visitConfigKeyValuePair(MemgraphCypher::ConfigKeyValuePairContext *ctx) {
+  MG_ASSERT(ctx->literal().size() == 2);
+  return std::pair{ctx->literal(0)->accept(this).as<Expression *>(), ctx->literal(1)->accept(this).as<Expression *>()};
+}
+
+antlrcpp::Any CypherMainVisitor::visitConfigMap(MemgraphCypher::ConfigMapContext *ctx) {
+  std::unordered_map<Expression *, Expression *> map;
+  for (auto *key_value_pair : ctx->configKeyValuePair()) {
+    // If the queries are cached, then only the stripped query is parsed, so the actual keys are cannot be determined
```

Comment:
I am happy to elaborate here, but I intentionally kept the comment short.
Basically when a query is parsed in case of the queries are cached, the first step is "stripping" the query, and the caching is based on stripped queries. During stripping, the query is transformed in a way that white-spaces (use only single white-spaces), literals (one value per literal type), casing and many other things are unified. Multiple queries result in the same stripped query, therefore the plan can be reused for them.

For further read please check [`ParseQuery` in cypher_query_interpreter.cpp](https://github.com/memgraph/memgraph/blob/bb1308acc7faeb010764ec06eeb587847bc47610/src/query/cypher_query_interpreter.cpp#L26-L30) and [`stripped.hpp`](https://github.com/memgraph/memgraph/blob/master/src/query/frontend/stripped.hpp).

Reply:
Wow thank you so much for the explanation! This perfect sense. (I thought that queries are cached the way they are, but it makes sense to strip them because as you say multiple queries result in the same stripped query -- and probably you can do all sorts of optimizations here beyond the scope of this discussion). 

Thank you so much for this <3